src/libs/routes/validationHandler.ts

```diff
@@ -0,0 +1,81 @@
+export default config => (req, res, next) => {
+  console.log("Inside Middleware");
+  const keys = Object.keys(config);
+
+  keys.forEach(key => {
+    const item = config[key];
+    console.log("Items are ", item);
+    let values = item.in.map(item => {
+      console.log("REQUEST---", req[item][key]);
+      return req[item][key];
+    });
+    console.log("Values are", values);
+    if (item && item.required) {
+      const validatedValues = values.filter(item => item);
+      if (validatedValues.length != values.length) {
+        console.log("ERROR----");
+        next({
+          status: "Bad Request",
+          message: item.errorMessage || `${key} is required` || "Error Message"
+        });
+      }
+      Object.keys(values).map(function(key) {
+        if (item.string && !(typeof values[key] == "string")) {
+          console.log(`${validatedValues} is not String`);
+          next({
+            status: "Bad Request",
+            message:
+              item.errorMessage ||
+              `${validatedValues} is not String` ||
+              "Error Message"
+          });
+        }
+        return values[key];
+      });
+      if (
+        item.regex &&
+        !new RegExp(item.regex).test(validatedValues.toString())
```

Comment:
validatedValues is an array. How can you directly change it to string. You need to apply the checks same as applied for required.